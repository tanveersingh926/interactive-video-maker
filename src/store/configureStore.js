import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import { myVideosSaga } from "./sagas/";
import { videoConfigurationReducer, myVideosReducer } from "./reducers";

const sagaMiddleware = createSagaMiddleware();
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  combineReducers({
    videoConfiguration: videoConfigurationReducer,
    myVideos: myVideosReducer
  }),
  composeEnhancer(applyMiddleware(sagaMiddleware))
);

const rootSagas = [myVideosSaga];
rootSagas.forEach(sagaMiddleware.run);

export default store;
