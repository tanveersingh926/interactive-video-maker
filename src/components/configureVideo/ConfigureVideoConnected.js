import { connect } from "react-redux";
import ConfigureVideo from "./ConfigureVideo";
import { deleteInteraction, saveVideo } from "../../store/actions";

const mapStateToProps = state => {
  return {
    videoTitle: state.videoConfiguration.videoTitle,
    videoId: state.videoConfiguration.videoId,
    shortDescription: state.videoConfiguration.shortDescription,
    interactions: state.videoConfiguration.interactions,
    videoDetails: state.videoConfiguration
  };
};

const mapDispatchToProps = dispatch => {
  return {
    deleteInteraction: interactionID => {
      dispatch(deleteInteraction(interactionID));
    },
    saveVideo: videoDetails => {
      dispatch(saveVideo(videoDetails));
    }
  };
};

const ConfigureVideoConnected = connect(
  mapStateToProps,
  mapDispatchToProps
)(ConfigureVideo);

export default ConfigureVideoConnected;
