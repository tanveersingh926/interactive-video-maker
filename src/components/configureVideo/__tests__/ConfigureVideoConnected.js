import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import ConfigureVideoConnected from "../ConfigureVideoConnected";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

const mockPushFn = jest.fn();
jest.mock("react-router-dom", () => ({
  useHistory: jest.fn(() => ({ push: mockPushFn })),
  useParams: jest.fn(() => ({ id: "youtubeId" }))
}));

document.getElementsByTagName = () => [
  {
    parentNode: {
      insertBefore: jest.fn()
    }
  }
];

const defaultInteraction = [
  {
    questionLabel: "Who is fastest man on eath?",
    questionId: "3845ff51-a2f5-40a4-a243-31aeb4025a62",
    responses: [
      {
        responseId: "bad0fea5-4ac8-426b-a42d-ff0a70721413",
        responseText: "req1",
        isCorrect: true,
        noOfTimesResponseSelected: 0,
        shouldOpenUrl: false,
        resumeVideo: true
      },
      {
        responseId: "61a5eaa5-587f-4a6a-a8ec-9102ba8b5d37",
        responseText: "res 2",
        noOfTimesResponseSelected: 0,
        shouldOpenUrl: false,
        resumeVideo: true
      }
    ],
    displayQuestionAtSecond: "04"
  }
];

const deleteInteraction = jest.fn();
const saveVideo = jest.fn();

const defaultProps = {
  videoTitle: "Title of Video",
  videoId: "youtubeId",
  shortDescription: "Short description of video",
  interactions: defaultInteraction,
  videoDetails: { details: "here" }
};
describe("component should render with all basic sections", () => {
  afterEach(cleanup);
  let store;
  beforeEach(() => {
    store = mockStore({
      videoConfiguration: { ...defaultProps }
    });
    store.dispatch = jest.fn();
  });
  const renderSubject = props => {
    const newProps = {
      deleteInteraction,
      saveVideo,
      ...props
    };

    return render(
      <Provider store={store}>
        <ConfigureVideoConnected {...newProps} />
      </Provider>
    );
  };

  it("should save snapshot of the page", async () => {
    const { asFragment } = renderSubject();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should click save video ", () => {
    const { getByText } = renderSubject();
    const saveBtn = getByText("Save Video");
    fireEvent.click(saveBtn);
    expect(store.dispatch).toHaveBeenCalledWith({
      payload: {
        interactions: [
          {
            displayQuestionAtSecond: "04",
            questionId: "3845ff51-a2f5-40a4-a243-31aeb4025a62",
            questionLabel: "Who is fastest man on eath?",
            responses: [
              {
                isCorrect: true,
                noOfTimesResponseSelected: 0,
                responseId: "bad0fea5-4ac8-426b-a42d-ff0a70721413",
                responseText: "req1",
                resumeVideo: true,
                shouldOpenUrl: false
              },
              {
                noOfTimesResponseSelected: 0,
                responseId: "61a5eaa5-587f-4a6a-a8ec-9102ba8b5d37",
                responseText: "res 2",
                resumeVideo: true,
                shouldOpenUrl: false
              }
            ]
          }
        ],
        shortDescription: "Short description of video",
        videoDetails: { details: "here" },
        videoId: "youtubeId",
        videoTitle: "Title of Video",
        isEmailRequired: false
      },
      type: "Save video to myVideos"
    });
  });

  it("should clicked on delete interaction", () => {
    const { getByTestId } = renderSubject();
    const deleteBtn = getByTestId("deleteInteraction");
    fireEvent.click(deleteBtn);
    expect(store.dispatch).toHaveBeenCalledWith({
      payload: "3845ff51-a2f5-40a4-a243-31aeb4025a62",
      type: "Delete interaction from the video via ID"
    });
  });
});
