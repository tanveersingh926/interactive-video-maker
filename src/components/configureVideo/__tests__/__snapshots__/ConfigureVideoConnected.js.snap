// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render with all basic sections should save snapshot of the page 1`] = `
<DocumentFragment>
  <div
    class="mt-4"
  >
    <h1
      class="mb-5"
    >
      Title of Video
    </h1>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <form
          class=""
        >
          <div
            class="form-row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="videoTitle"
                >
                  Title
                </label>
                <input
                  class="form-control"
                  disabled=""
                  id="videoTitle"
                  name="title"
                  placeholder="Enter Video Title"
                  type="text"
                  value="Title of Video"
                />
              </div>
            </div>
            <div
              class="col-md-6"
            >
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="videoShortDescription"
                >
                  Short Description
                </label>
                <input
                  class="form-control"
                  disabled=""
                  id="videoShortDescription"
                  name="shortDescription"
                  placeholder="Enter Short Description"
                  type="text"
                  value="Short description of video"
                />
              </div>
            </div>
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
              for="videoShareUrl"
            >
              Share URL
            </label>
            <input
              class="form-control"
              id="videoShareUrl"
              name="shareUrl"
              readonly=""
              type="text"
              value="http://localhost/video/youtubeId/share"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
              for="videoEmbed"
            >
              Embed
            </label>
            <textarea
              class="form-control"
              id="videoEmbed"
              name="embed"
            >
              &lt;iframe src="http://localhost/video/youtubeId/embed" allowtransparency="true" frameborder="0" scrolling="no" width="640" height="390"&gt;&lt;/iframe&gt;
            </textarea>
          </div>
          <button
            class="mt-4 mr-2 btn btn-primary"
            type="submit"
          >
            Save Video
          </button>
          <button
            class="mt-4 btn btn-primary"
            type="button"
          >
            Add Interaction
          </button>
        </form>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="modal-inline--container text-left clearfix modal-inline__not-standalone"
        >
          <div
            id="youtube-player-youtubeId"
          >
             
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="mt-5"
        >
          Interactions
        </h3>
        <ul
          class="mb-4 list-group"
        >
          <li
            class="list-group-item"
          >
            1. Who is fastest man on eath? - 
            <span
              class="text-primary"
            >
              00:04
            </span>
            <span
              class="float-right"
            >
              <button
                aria-label="Close"
                class="close text-danger"
                data-testid="deleteInteraction"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
